nixos/modules/services/web-servers/webhook.nix

```
@@ -0,0 +1,89 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.webhook;
+  webhook = pkgs.webhook;
+  user = if (cfg.user == null) then "webhook" else cfg.user;
+  group = if (cfg.group == null) then "webhook" else cfg.group;
+  settingsFormat = pkgs.formats.json {};
+  configFile = settingsFormat.generate "webhook.json" cfg.settings;
+
+in {
+  options.services.webhook = {
+
+    enable = mkEnableOption "the webhook service";
+
+    user = mkOption {
+      default = null;
+      description = "User webhook service runs as. If null, it will default to `webhook`";
+      type = with types; nullOr str;
+    };
+
+    group = mkOption {
+      default = null;
+      description = "Group webhook service runs as. If null, it will default to `webhook`";
+      type = with types; nullOr str;
+    };
+
+    settings = mkOption {
+      description = "Webhook service settings.";
+      type = settingsFormat.type;
+      example = literalExpression ''
+        [{
+          id = "redeploy-webhook";
+          execute-command = "/var/scripts/redeploy.sh";
+          command-working-directory = "/var/webhook";
+        }]
+      '';
+    };
+
+    path = mkOption {
+      default = [];
+      description = "Which packages the service can access";
+      example = literalExpression ''
+        with pkgs; [ nixFlakes ]
+      '';
+      type = with types; listOf package;
+    };
+
+    extraOptions = mkOption {
+      default = "-verbose";
+      description = "Which flags to append to the command";
+      example = "-debug -port 12345";
+      type = types.str;
```

this should be a list to enable merging.
```suggestion
      type = types.listOf types.str;
```